### ISSUE TITLE
How to set up and use HAProxy

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#4388]
We cover HAproxy in our on-premise deployment tutorial, but we really need a stand-alone tutorial on properly setting up HAProxy. This should cover configs in more detail (see https://github.com/cockroachdb/docs/issues/4387) as well as using multiple processes (is single-threaded by default) to speed things up.

**Comment by awoods187** [comment]
From a customer: Customer was asking about how to setup HAProxy so that it would use the local servers first and then make use of the remote servers next if the local ones are not available. This seems to be possible by using the option backup on the server line along with the option allbackups. We should document this as it lines up with how we recommend Cockroach to be deployed across regions.

https://github.com/cockroachlabs/support/issues/272

**Comment by awoods187** [comment]
Another customer linked this doc to us https://www.haproxy.com/blog/dynamic-configuration-haproxy-runtime-api/

**Comment by awoods187** [comment]
@jseldess this might be a good project for 19.2

**Comment by jseldess** [comment]
I agree, @awoods187. @lhirata, @rolandcrosby, not sure how this weighs against other priorities in ops and tools, but please consider.

**Comment by mwang1026** [comment]
also seems deployments and/or server @piyush-singh

**Comment by piyush-singh** [comment]
cc @taroface and @johnrk - this sounds to me like a deployments and ops docs project. Let me know if you agree - will leave it to you to prioritize as you see fit if so. If not, I can triage and find a home for this.

**Comment by jseldess** [comment]
Definitely agree it's deploy and ops. We leverage HAProxy in our on-prem deployment tutorial.
